develop/tutorials/articles/28-themes-and-layout-templates/xx-creating-forms-with-liferays-taglibs.markdown

```diff
@@ -0,0 +1,229 @@
+                    </aui:fieldset>
+            </aui:fieldset-group>
+
+            <aui:button-row>
+                    <aui:button type="submit" />
+
+                    <aui:button href="<%= redirect %>" type="cancel" />
+            </aui:button-row>
+    </aui:form>
+
+As you can see, a standard form uses quite a few taglibs. If you take a closer
+look, you'll notice that all of the taglibs are prefixed with `aui`. `aui`
+stands for [AlloyUI](http://alloyui.com/), a JavaScript framework that uses
+Bootstrap to allow you to create UI components, easily and effectively. In
+order to use these AUI tags, you will need to have the AUI taglib declaration
+imported into your JSP. You'll take care of this next.
+
+## Adding the Taglib Declaration [](id=adding-the-taglib-declaration)
+
+The first thing you'll need to do is make sure you have the
+`<%@ taglib uri="http://liferay.com/tld/aui" prefix="aui"%>` declaration is in
+your JSP. This can be added to the main view of your app, `view.jsp` for example,
+or it can be added to a separate JSP file, such as `init.jsp`, and imported into
+the JSP in which you want to show the form. For example, the Portlet
+Configuration application includes the taglib declarations in its `init.jsp`,
+and then imports the `init.jsp` into its `view.jsp`, using the following line:
+
+    <%@ include file="/init.jsp"%>
+
+Once the AUI taglib declaration is imported, you can move onto creating the form
+next.
+
+## Creating the Form [](id=creating-the-form)
+
+A form's design is determined by the input needed from the user. To that end,
+there are multiple design possibilities. The examples covered in the sections
+that follow are one possible design, and highlight some of the available
+attributes for the tags. The steps are laid out in a natural order, however you
+can jump to any section you wish.
+
+For a full list of the available attributes for the form tags covered, checkout
+the API docs for the [AUI Tags](https://docs.liferay.com/portal/7.0/taglibs/util-taglib/aui/tld-summary.html).
+
+Go ahead and get started by adding the form tag next.
+
+### Adding the Form Wrapper [](id=adding-the-form-wrapper)
+
+Start off by adding a `<aui:form>` tag to your jsp. Make sure to add a closing
+`<aui:form/>` tag, to wrap your form. This acts as a wrapper for your form and
+offers some additional styling and custom portlet namespacing for you to use. If
+you are familiar with HTML `<form>` tags, `<aui:form>` tags are configured the
+same way. For instance, the Bloggs Aggregator app has the following
+configuration:
+
+    <aui:form action="<%= configurationActionURL %>" method="post" name="fm">
+
+The `action` attribute specifies where to send the form data when the form is
+submitted. The `method` attribute defines the method to use to send the form
+data(in most cases this will be post). Finally, the `name` attibute specifies
+the name for the form, as well as the ID for the component instance for the
```

Comment:
@mwilliams2014 Maybe it's worth mentioning that by default aui:form will put the portlet namespace in front of the name and id attributes?